### ISSUE TITLE
Assign macroinvertebrate taxa to new pollution sensitivity groups

### ISSUE THREAD

**Issue by HeatherBrooks** [ODM2/ODM2DataSharingPortal#406]
**UPDATE**, this list below is not exactly correct. Please reference the  [Leaf Pack Network Macroinvertebrate ITIS Identifiers-2019](https://docs.google.com/spreadsheets/d/1KDY1lx4A5zW-6jlLcv_YrS8MrrjNZredL_QijpJct9I/edit?usp=sharing) Google Sheet for exact values for TaxonName (=scientific name) and CommonName.

New pollution sensitivity groups and weighting values should be assigned as follows:
 - Pollution Sensitivity Group 1 (label=”Group 1: Sensitive”; weighting value = 3)
    - Stoneflies (Plecoptera)
    - Mayflies (Ephemeroptera)
    - Other caddisflies (Trichoptera) UPDATED 2019-12-18: label should read "Non-net-spinning caddisflies (Trichoptera other than Hydropsychidae)"
    - Dobsonflies, fishflies, and alderflies (Megaloptera)
    - Riffle beetles larvae/adults (Elmidae)
    - Water pennies (Psephenidae)
    - Right-handed/gilled snails (Gastropoda)
    - Aquatic snipe flies (Athericidae)
- Pollution Sensitivity Group 2 (label=”Group 2: Somewhat Sensitive”; weighting value = 2)
    - Damselflies (Odonata)
    - Dragonflies (Odonata)
    - Aquatic sowbugs (Isopoda)
    - Scuds (Amphipoda)
    - Crane flies (Tipulidae)
    - Clams/mussels (Mollusca)
    - Crayfish (Decapoda)
    - Net-spinning caddisflies (Hydropsychidae)
- Pollution Sensitivity Group 3 (label=”Group 3: Tolerant”; weighting value = 1)
    - Midge flies (Chironomidae)
    - Black flies (Simuliidae)
    - Planarians (Turbellaria)
    - Leeches (Hirudinea)
    - Left-handed/lunged snails (Gastropoda)
    - Aquatic worms (Oligochaeta)
    - Rat-tailed maggots (Syrphidae)

**Comment by htaolimno** [comment]
@HeatherBrooks  is there any order to display the taxa in each group ? they look random to me.

**Comment by HeatherBrooks** [comment]
@htaolimno The taxa display order in each group needs to match the order in the [Experiment Summary Data Sheet](https://drive.google.com/open?id=1azne5h8DDHKLYV73N1ypbVq8Da7CL4ml) included in the Leaf Pack Experiment kit. We're trying to maintain consistency between the paper forms people use to record their data during their experiment and the portal data entry form. The correct order is also listed in the [original post](https://github.com/ODM2/ODM2DataSharingPortal/issues/406#issue-462162879) in this issue. (Note that I have edited the original post this morning to reflect small changes in common names since I posted the issue in June.)

**Comment by htaolimno** [comment]
@HeatherBrooks sounds good. thank you.

**Comment by aufdenkampe** [comment]
@htaolimno just deployed an update on this issue to staging. For an example, see https://staging.monitormywatershed.org/sites/StroudEduc_TEST/leafpack/18/.

However, I see that there was some confusion between the guidance provided in this issue (above at https://github.com/ODM2/ODM2DataSharingPortal/issues/406#issue-462162879), versus in https://github.com/ODM2/ODM2DataSharingPortal/issues/404#issuecomment-539565688 and the discussion following it.

@htaolimno, as you'll see in https://github.com/ODM2/ODM2DataSharingPortal/issues/404#issuecomment-541888124, @HeatherBrooks had updated [Leaf Pack Network Macroinvertebrate ITIS Identifiers-2019](https://docs.google.com/spreadsheets/d/1KDY1lx4A5zW-6jlLcv_YrS8MrrjNZredL_QijpJct9I/edit?usp=sharing) on Oct. 14 with some name changes. Specifically:
-  "Other caddisflies (Trichoptera)" should be changed to "Non-net-spinning caddisflies (Trichoptera other than Hydropsychidae)"
- "Rat-tailed maggots (Syrphidae)" should be changed to "Rat-tailed maggots (Eristalinae)"

@HeatherBrooks, can you confirm?

**Comment by HeatherBrooks** [comment]
@aufdenkampe @htaolimno 
- "Other caddisflies (Trichoptera)" should be changed to "Non-net-spinning caddisflies (Trichoptera other than Hydropsychidae)" PENDING Sorry, need to verify again with Arscott as this deviates from leaf pack kit data entry form.
- "Rat-tailed maggots (Syrphidae)" should be changed to "Rat-tailed maggots (Eristalinae)" INCORRECT. My mistake -- I took it to subfamily in ITIS -- it should remain as Syrphidae

**Comment by HeatherBrooks** [comment]
@aufdenkampe @htaolimno I have confirmed: Other caddisflies (Trichoptera)" should be changed to "Non-net-spinning caddisflies (Trichoptera other than Hydropsychidae)

**Comment by HeatherBrooks** [comment]
@aufdenkampe @htaolimno I've checked all task assigned in the original post against [https://staging.monitormywatershed.org/sites/StroudEduc_TEST/leafpack/18/](url). Everything will be complete once the "Other caddisflies (Trichoptera)" label is changed to "Non-net-spinning caddisflies (Trichoptera other than Hydropsychidae)"

**Comment by aufdenkampe** [comment]
@HeatherBrooks, thanks for all this.

We'll be using the [Leaf Pack Network Macroinvertebrate ITIS Identifiers-2019](https://docs.google.com/spreadsheets/d/1KDY1lx4A5zW-6jlLcv_YrS8MrrjNZredL_QijpJct9I/edit?usp=sharing) Google Sheet as our reference for TaxonName (scientific) and CommonName.

Can you confirm that these names are correct? I think I have them right.

**Comment by HeatherBrooks** [comment]
@aufdenkampe I've just rechecked [Leaf Pack Network Macroinvertebrate ITIS Identifiers-2019](https://docs.google.com/spreadsheets/d/1KDY1lx4A5zW-6jlLcv_YrS8MrrjNZredL_QijpJct9I/edit?usp=sharing). TaxonName and CommonName are correct as listed. 

However, please note that for five taxa we are asking that the Parent Taxon rather than the TaxonName be displayed in the data entry form. They should be listed as:

- Right-handed/gilled snails (Gastropoda)
- Left-handed/lunged snails (Gastropoda)
- Damselflies (Odonata)
- Dragonflies (Odonata)
- Clams/mussels (Mollusca)

These changes were requested in #404, but I wanted to make sure they were clear here. I've also added them as comments in the TaxonName cells in the Google Sheet.

**Comment by aufdenkampe** [comment]
@HeatherBrooks, I just told @htaolimno that the use of ParentTaxon was a mistake for those 5 entries. I understand your need to match the sheet, but from a data management perspective, it's important that each of these entries refers to something unique. Can I suggest for these five that we use the general syntax of:
- Common Name (ParentTaxon, TaxonName)

Specifically, 
- Right-handed/gilled snails (Gastropoda, Prosobranchia)
- Left-handed/lunged snails (Gastropoda, Archaeopulmonata)
- Damselflies (Odonata, Zygoptera)
- Dragonflies (Odonata, Anisoptera)
- Clams/mussels (Mollusca, Bivalvia)

**Comment by HeatherBrooks** [comment]
@aufdenkampe @htaolimno  It is important to have uniqueness in the labels in the user interface? We thought that by [retaining the current database entries](https://sites.google.com/a/stroudcenter.org/intranet/directory) (Prosobranchia, Archaeopulmonata, Zygoptera, etc.) and just changing the user-facing labels we would be okay from a database management perspective.

**Comment by aufdenkampe** [comment]
@HeatherBrooks, it is true that we can work around the uniqueness constraint on the User Interface. 

I do understand your need to have some match with the LPN data worksheets, and I was thinking that my solution also provides that in addition to meeting additional needs.

There are several reasons why I feel rather strongly that we need a 1:1 correspondence between the common name and the scientific name (ParentTaxon, TaxonName) in the parentheses, and also some consistent logic between the UI and the backend database. The reasons include:
- We want to accommodate users who do not have the official LPN worksheets in front of them (I already know of a group in MN), which is one of the reasons we added the ITIS links.
- We want the link to the ITIS record to match the scientific name
- We want to accommodate knowledgable users who know these scientific names, and not confuse them.
- We want the UI to present consistency in the pattern of how macroinvertebrates are listed, not only to help knowledgable users but also to help future developers (one of the challenges we've found in getting up to speed on the code are inconsistent naming of variables & code objects, which we're interested in cleaning up over time).

In short, having the common name consistently match the scientific name demonstrates professionalism.

**Comment by HeatherBrooks** [comment]
Gotcha. Please implement the Common Name (ParentTaxon, TaxonName) scheme for the five troublesome taxa.

**Comment by aufdenkampe** [comment]
This is now complete with commit 87d05d304474d7eee3d3567e1b57d92cc57ca90c.
@HeatherBrooks, does it look good? If so, please close this issue.

**Comment by aufdenkampe** [comment]
I'll save you the trouble of closing!

**Comment by HeatherBrooks** [comment]
@aufdenkampe Looks like there's one little glitch:
Now reads: Riffle beetles larvae/adults (Elmidae-adults and larvae)
Should read: Riffle beetles larvae/adults (Elmidae)

**Comment by aufdenkampe** [comment]
Thanks for catching that!
@htaolimno, can you make that edit described above by @HeatherBrooks.

**Comment by aufdenkampe** [comment]
@htaolimno, can you make this minor change to the scientific name that @HeatherBrooks mentions in https://github.com/ODM2/ODM2DataSharingPortal/issues/406#issuecomment-568081262?

In short, please change "(Elmidae-adults and larvae)" to "(Elmidae)".
We only want to mention "larvae/adults" in the common name.

**Comment by htaolimno** [comment]
I've made the change to the scientific name from "Elmidae-adults and larvae" to "Elmidae" in the staging database. Will run the same SQL statement on production server when we are ready to publish the changes to production.

**Comment by htaolimno** [comment]
@HeatherBrooks ready for testing. Thank you.

**Comment by HeatherBrooks** [comment]
@htaolimno Looks good, thanks.
